### PR TITLE
Clean up Steensgaard alias analysis

### PR DESCRIPTION
**Pull request by phate**
This PR cleans up the Steensgaard alias analysis. Specifically, it does the following:
1. Align layout of Steensgaard.hpp and Steensgaard.cpp
2. Move LocationSet class to Steensgaard.cpp
3. Discard the internal state after we are done with the analysis to free up memory
4. Rename some private methods for clarity
5. Move join method from Steensgaard class to LocationSet class

### CODE REVIEW

File: jlm/llvm/opt/alias-analyses/Steensgaard.cpp [hunk 1]
```diff
@@ -561,234 +473,405 @@ class DummyLocation final : public Location {
...
-  for (auto & set : DisjointLocationSet_) {
-    str += dot_node(set) + "\n";
+      return jlm::util::strfmt("{ ", (intptr_t)&set, " [label = \"", setLabel, "\"]; }");
+    };
 
-    auto pointsTo = set.value()->GetPointsTo();
-    if (pointsTo != nullptr) {
-      auto pointsToSet = DisjointLocationSet_.find(pointsTo);
-      str += dot_edge(set, *pointsToSet) + "\n";
+    auto dot_edge = [&](const DisjointLocationSet::set & set, const DisjointLocationSet::set & pointsToSet)
```
Comment (sjalander) [inline jlm/llvm/opt/alias-analyses/Steensgaard.cpp:662]: shouldn't it be dotEdge?

File: jlm/llvm/opt/alias-analyses/Steensgaard.cpp [hunk 2]
```diff
@@ -561,234 +473,405 @@ class DummyLocation final : public Location {
...
+    std::string str;
+    str.append("digraph PointsToGraph {\n");
+
+    for (auto & set : DisjointLocationSet_) {
+      str += dot_node(set) + "\n";
+
+      auto pointsTo = set.value()->GetPointsTo();
+      if (pointsTo != nullptr) {
+        auto pointsToSet = DisjointLocationSet_.find(pointsTo);
+        str += dot_edge(set, *pointsToSet) + "\n";
```
Comment (sjalander) [inline jlm/llvm/opt/alias-analyses/Steensgaard.cpp:676]: dotEdge()

File: jlm/llvm/opt/alias-analyses/Steensgaard.cpp [hunk 3]
```diff
@@ -561,234 +473,405 @@ class DummyLocation final : public Location {
...
+    auto it = LocationMap_.find(&output);
+    return it == LocationMap_.end() ? nullptr : it->second;
+  }
 
-  std::string str;
-  str.append("digraph PointsToGraph {\n");
+  std::string
+  ToDot() const
+  {
+    auto dot_node = [](const DisjointLocationSet::set & set)
```
Comment (sjalander) [inline jlm/llvm/opt/alias-analyses/Steensgaard.cpp:636]: dotNode

**Review by sjalander — APPROVED** [review]

File: jlm/llvm/opt/alias-analyses/Steensgaard.cpp (hunk 1 above)
Comment (phate) [reply jlm/llvm/opt/alias-analyses/Steensgaard.cpp]: Fixed.

File: jlm/llvm/opt/alias-analyses/Steensgaard.cpp (hunk 2 above)
Comment (phate) [reply jlm/llvm/opt/alias-analyses/Steensgaard.cpp]: Fixed.

File: jlm/llvm/opt/alias-analyses/Steensgaard.cpp (hunk 3 above)
Comment (phate) [reply jlm/llvm/opt/alias-analyses/Steensgaard.cpp]: Fixed.

**Pull request merged**